packages/experimental/src/components/BreadcrumbWithOverflow/BreadcrumbWithOverflow.js

```diff
@@ -71,124 +71,166 @@ export const BreadcrumbWithOverflow = ({
+    //  * the first item is the next to be displayed, if there's space once the last item and overflow are shown;
+    //  * any remaining space after the first item, last item and overflow are shown is used to show items (n-1), (n-2), (n-3), ..., until the space is used up ;
+    // Note that displayCount (min 1) has been computed based on the available space and the above sequence.
+    const overflowStart = displayCount > 1 ? 1 : 0;
+    // Everything but the last item if 1, otherwise 1 to length + overflowStart - displayCount
+    for (
+      let i = overflowStart;
+      i < childArray.length + overflowStart - displayCount;
+      i++
+    ) {
+      // add items into the overflow menu
+      child = childArray[i];
+
       newOverflowBreadcrumbItems.push(
         React.cloneElement(child, {
-          key: `displayed-breadcrumb-${internalId}-overflow-item-${index}`,
+          key: `displayed-breadcrumb-${internalId}-overflow-item-${i}`,
         })
       );
     }
 
-    if (displayCount === 0) {
+    // add the first item before overflow if space
+    if (displayCount > 1) {
+      newDisplayedBreadcrumbItems.push(
+        React.cloneElement(childArray[0], {
+          className: cx([
+            childArray[0].props.className,
+            `${blockClass}--displayed-breadcrumb`,
+          ]),
+          key: `displayed-breadcrumb-${internalId.current}-0`,
+        })
+      );
+    }
+
+    // if needed add overflow menu after first item or before last
+    if (displayCount < childArray.length) {
       newDisplayedBreadcrumbItems.push(
         <BreadcrumbOverflowMenu
           overflowItems={newOverflowBreadcrumbItems}
           key={`$displayed-breadcrumb-${internalId}-overflow`}
         />
       );
-    } else {
-      let displayed = 0;
-      for (
-        let index = 0;
-        displayed < displayCount && index < childArray.length;
-        index++
-      ) {
-        displayed++;
-
-        if (index === 1 && displayCount < childArray.length) {
-          // we only want the last childArray.length - displayCount
-          index += childArray.length - displayCount;
-        }
+    }
 
-        child = childArray[index];
-        newDisplayedBreadcrumbItems.push(
-          React.cloneElement(child, {
-            className: cx([
-              child.props.className,
-              `${blockClass}--displayed-breadcrumb`,
-            ]),
-            key: `displayed-breadcrumb-${internalId.current}-${index}`,
-          })
-        );
-        if (index === 0 && displayCount < childArray.length) {
-          // add overflow menu after first item
-          newDisplayedBreadcrumbItems.push(
-            <BreadcrumbOverflowMenu
-              overflowItems={newOverflowBreadcrumbItems}
-              key={`$displayed-breadcrumb-${internalId}-overflow`}
-            />
-          );
-        }
-      }
+    // skip items in overflow + 1 for first item
+    for (
+      let i = newOverflowBreadcrumbItems.length + 1;
```

Comment:
I don't think this works for a one-item breadcrumb: it will not insert the final (and only) breadcrumb item.
```suggestion
      let i = newOverflowBreadcrumbItems.length + overflowStart;
```